src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs

```diff
@@ -45,7 +45,7 @@ public override void ExecuteResult([NotNull] ActionContext context)
         {
             var urlHelper = GetUrlHelper(context);
 
-            var destinationUrl = urlHelper.RouteUrl(RouteValues);
+            var destinationUrl = urlHelper.RouteUrl(RouteName, RouteValues);
```

Comment:
Functional tests are harder to debug and take more time to write. They are great when we want to capture a user-focused scenario or complex integration point.